components/brave_wallet/browser/solana_instruction_decoded_data_unittest.cc

```diff
@@ -0,0 +1,147 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/solana_instruction_decoded_data.h"
+
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "base/test/gtest_util.h"
+#include "base/test/values_test_util.h"
+#include "brave/components/brave_wallet/browser/solana_instruction_data_decoder.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+TEST(SolanaInstructionDecodedDataUnitTest, FromToMojom) {
+  std::vector<mojom::SolanaInstructionParamPtr> mojom_params;
+  mojom_params.emplace_back(
+      mojom::SolanaInstructionParam::New("test1", "Test1", "TEST1"));
+  mojom_params.emplace_back(
+      mojom::SolanaInstructionParam::New("test2", "Test2", "TEST2"));
+  auto mojom_decoded_data = mojom::DecodedSolanaInstructionData::New(
+      static_cast<uint32_t>(mojom::SolanaSystemInstruction::kTransfer),
+      solana_ins_data_decoder::GetMojomAccountParamsForTesting(
+          mojom::SolanaSystemInstruction::kTransfer, absl::nullopt),
+      std::move(mojom_params));
+
+  auto decoded_data = SolanaInstructionDecodedData::FromMojom(
+      mojom::kSolanaSystemProgramId, mojom_decoded_data.Clone());
+  ASSERT_TRUE(decoded_data);
+  EXPECT_EQ(decoded_data->sys_ins_type,
+            mojom::SolanaSystemInstruction::kTransfer);
+  EXPECT_FALSE(decoded_data->token_ins_type);
+  std::vector<std::tuple<std::string, std::string, std::string>> expect_params(
+      {{"test1", "Test1", "TEST1"}, {"test2", "Test2", "TEST2"}});
+  EXPECT_EQ(decoded_data->params, expect_params);
+  EXPECT_EQ(decoded_data->ToMojom(), mojom_decoded_data);
+
```

Comment:
shouldn't we also check account params?

Reply:
Yes, totally missed it. https://github.com/brave/brave-core/pull/15142/commits/44ea0402ec1fff92734c8cb2f017f0806ac04158

Reply:
we also need to check it for `kApprove` down below

Reply:
https://github.com/brave/brave-core/pull/15142/commits/732d8ad8cb3af672926b7da98628f276d3440d17